tests: shell: migrate tests to the new shell

jakub-uC opened the pull request:
Testing:
1. Shell built-in commands.
2. Wildcards module.
3. Static commands creation and execution.
4. Dynamic commands creation and execution.

I have removed: `tests/subsys/fs/multi-fs/prj_fs_shell.conf` because this project can be only build for qemu and qemu does not have `DT_FLASH_DEV_NAME`

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

jarz-nordic commented:
Blocked by: PR#11175

jarz-nordic commented:
@aescolar : May you please take a look at tests? Tests are failing on posix, I have no idea how to debug it.

aescolar commented:
@jarz-nordic: quite a few tests are not compiling in quite a few platforms. At a first glance, this does not seem a posix arch related issue.

aescolar commented on tests/include/tc_util.h:
```diff
@@ -12,7 +12,7 @@
 
 #include <string.h>
-#include <shell/legacy_shell.h>
+#include <shell/shell.h>
```
All unit tests are failing because of this..
```
In file included from include/logging/log.h:11:0,
                 include/shell/shell.h:16,
                 tests/include/tc_util.h:15,
                 tests/ztest/include/ztest.h:62,
                 tests/ztest/src/ztest.c:7:
include/logging/log_core.h:19:2: error: #error "Logger does not support 64 bit architecture."
 #error "Logger does not support 64 bit architecture."
```
Unit tests are compiled fully natively (there is no posix arch involved)

jarz-nordic commented:
@aescolar : Thank you very much for this investigation.
Regarding select serial I am confused how to fix that :/ It was implemented as a result of these discussions:
https://github.com/zephyrproject-rtos/zephyr/pull/10379#discussion_r222674291

aescolar commented:
@jarz-nordic : As @ulfalizer usually reminds, `select` is a dangerous thing to use unless dependencies are quite trivial and well understood.
I would think it could be reasonable to have `SHELL_BACKEND_SERIAL` depend on `SERIAL`, then you should be able to have it `default to y` as it would only default if SERIAL is available.
The argument in #10379 from Jukka was about not needing to change the samples/tests config. But I guess you could change the boards' defconfig files to have a `if SHELL\n config SERIAL\n default y\n endif` for boards which do have SERIAL.
Again, I think it is better if @ulfalizer chips in.

jarz-nordic commented:
Blocked by #11532

varun-sha commented on tests/subsys/fs/multi-fs/testcase.yaml:
```diff
@@ -2,7 +2,3 @@ tests:
     platform_whitelist: qemu_x86
     tags: nffs fatfs filesystem
-  filesystem.fs_shell:
-    extra_args: CONF_FILE="prj_fs_shell.conf"
```
Both tests of fs_shell & multifs use qemu_x86 as platform whitelist, how come only "prj_fs_shell.conf" is causing issues and not prj.conf?

I am also referring below comment:
I am going to remove tests/subsys/fs/multi-fs/prj_fs_shell.conf because this project can be only build for qemu and qemu does not have DT_FLASH_DEV_NAME.

Commit:
test: unittest: run for 32bit architecture

Running tests compiled for 64bit architecture were causing errors
in logger module. Updated unittest.cmake to build tests only
for 32bit architecture.

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

Commit:
tests: shell: migrate tests to the new shell

Testing:
1. Shell built-in commands.
2. Wildcards module.
3. Static commands creation and execution.
4. Dynamic commands creation and execution.

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>
